### PR TITLE
[FIX] analytic: fix analytic group issue

### PR DESCRIPTION
**Pull request by kcv-odoo**
Steps:
- Install Expense app.
- Enable analytic account and tags from user setting.
- Set analytic tags on a Expense.
- Disable analytic account.
- Try to open Expense app.

Issue:
- Can't able to open Expense app access error.

Cause:
- Wrong group on access rule of analytic account tags and
internal user does not have tags read access.

Fix:
- Set correct group on analytic account tags access rule and
give internal user read access.

task-2861376

### CODE REVIEW

**Review by qdp-odoo — CHANGES_REQUESTED** [review]

File: addons/analytic/security/ir.model.access.csv [hunk 1]
```diff
@@ -1,6 +1,7 @@
 id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
 access_account_analytic_account,access_account_analytic_account,model_account_analytic_account,group_analytic_accounting,1,1,1,1
 access_account_analytic_line,access_account_analytic_line,model_account_analytic_line,group_analytic_accounting,1,1,1,1
-access_account_analytic_tag,access_account_analytic_tag,model_account_analytic_tag,group_analytic_accounting,1,1,1,1
+access_account_analytic_tag,access_account_analytic_tag,model_account_analytic_tag,group_analytic_tags,1,1,1,1
+access_account_analytic_tag_user,access_account_analytic_tag_user,model_account_analytic_tag,base.group_user,1,0,0,0
```
Comment (xavierbol) [inline addons/analytic/security/ir.model.access.csv:5]: You don't really `hide that record if user have no access to tag.` so your commit message is not really right. 🤔

**Review by xavierbol — COMMENTED** [review]
I suggest to keep your commit message and instead of your changes, we will add the `groups='analytic.group_analytic_accounting'` in all `many2many` field in which the comodel is the `account.analytic.tag` model. By doing that we will hide the field when the user has not the group.

For instance in `project.project`, you have to change [this line](https://github.com/odoo/odoo/blob/8577184187851f1e90fa6fd73a176662460c6124/addons/project/models/project.py#L299) like this:
```diff
-    analytic_tag_ids = fields.Many2many('account.analytic.tag', string='Analytic Tags')
+    analytic_tag_ids = fields.Many2many('account.analytic.tag', string='Analytic Tags', groups='analytic.group_analytic_accounting')
```

**Comment by kcv-odoo** [comment]
> I suggest to keep your commit message and instead of your changes, we will add the `groups='analytic.group_analytic_accounting'` in all `many2many` field in which the comodel is the `account.analytic.tag` model. By doing that we will hide the field when the user has not the group.
> 
> For instance in `project.project`, you have to change [this line](https://github.com/odoo/odoo/blob/8577184187851f1e90fa6fd73a176662460c6124/addons/project/models/project.py#L299) like this:
> 
> ```diff
> -    analytic_tag_ids = fields.Many2many('account.analytic.tag', string='Analytic Tags')
> +    analytic_tag_ids = fields.Many2many('account.analytic.tag', string='Analytic Tags', groups='analytic.group_analytic_accounting')
> ```

https://nimb.ws/dNVtzi when analytic tags are set on project and try to create new task in that project when user has not analytic tags access then it'll try to access tags of project to set on new task in default_get same for timesheets.
Even if I add group in all 'account.analytic.tag'`s M2M fields still it'll not fix above issue. Should I have find another way to fix above issue rather then giving base user read access?

**Comment by xavierbol** [comment]
Even if the user has the read access the issue that you described is still there, isn't it? 🤔

**Comment by kcv-odoo** [comment]
Up to saas-15.3

**Comment by Levizar** [comment]
I'm closing the pr because:
- It's old
- No one seems to complain about this
- No answer given to Xavierbol
- It's not linked to the related task anymore and a R&D Feedback has been created

**Pull request closed without merging**